pallets/contracts/src/lib.rs

```diff
@@ -166,7 +177,8 @@ decl_module! {
         #[weight = 50_000_000.saturating_add(pallet_contracts::Call::<T>::put_code(code.clone()).get_dispatch_info().weight)]
         pub fn put_code(
             origin,
-            meta_info: SmartExtensionMetadata<BalanceOf<T>>,
+            meta_info: SmartExtTemplateMetadata<BalanceOf<T>>,
```

Comment:
Suggestion - maybe use `TemplateMetadata` instead of `SmartExtTemplateMetadata`.

In general I was thinking we could generalise the `contract` pallets naming for this stuff - it doesn't really care (or know) whether a WASM contract blob is a smart extension or not. As part of the metadata we can still capture the type (i.e. TransferExtension), but this is just set manually by the author anyway.

I like the idea of using `Template` instead (like you have with `TemplateFeesChanged` etc.) - we could change `ExtensionUsageFee` to `TemplateFee` and so on.

Reply:
`ExtensionUsageFee` always references to a smart extension instance so I think it is better to keep Extension prefix for this particular use case and for rest I think it makes sense to refer to the Template prefix instead of Smart extension prefix.